hpat/hiframes/boxing.py

```diff
@@ -155,10 +155,18 @@ def _infer_series_list_dtype(S):
 
 def _infer_index_type(index):
     # TODO: support proper inference
+
+    # np.dtype('O') - Python objects
     if index.dtype == np.dtype('O') and len(index) > 0:
         first_val = index[0]
         if isinstance(first_val, str):
             return string_array_type
+
+    elif index.dtype == np.int64:
+        # RangeIndex or Int64Index
+        dtype = numba.numpy_support.from_dtype(index.dtype)
+        return types.Array(dtype, 1, 'C')
+
     return types.none
```

Comment:
```suggestion
    return index.dtype
```
and no need code (elif) above?